src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs

```diff
@@ -165,7 +165,7 @@ public async Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int p
 
             var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
             var triggerText = sourceText.ToString(triggerToken.Span);
-            var fallbackOptions = _globalOptions.GetCodeCleanupOptionsProvider();
+            var fallbackOptions = _globalOptions.CreateProvider();
```

Comment:
hrmm.. this reads a bit odd.  what sort of provider is this making?

Reply:
This one: https://github.com/dotnet/roslyn/pull/60882/files/4394aa5ebde133b78ccfec65addba40bf7ddbf69#diff-0b999bc4c9feee2f0732a23368a04773ea9665f5b2f652144461123d6113366eR27

Yeah, it's not very descriptive. I'll rename to `CreateCodeActionOptionsProvider`. It's just temporary. the `IGlobalOptionsService` should itself implement the provider interfaces. Currently that's not possible due to layering, but eventually it should be.

Reply:
Actually, not sure about the rename to `CreateCodeActionOptionsProvider`. It creates a provider that implements various options. It's not just CodeActionOptions, but all kinds of other options. While CodeActionOptions generally have all of these options, it's misleading to call the factory `CreateCodeActionOptionsProvider` since the call site might not actually care about code action options but other options. So I think I'd rather keep the name as is for now.